mesa/space.py

```diff
@@ -549,10 +549,10 @@ def default_val() -> MultiGridContent:
     def place_agent(self, agent: Agent, pos: Coordinate) -> None:
         """Place the agent at the specified location, and set its pos variable."""
         x, y = pos
-        if agent not in self.grid[x][y]:
+        if agent.pos != pos or agent not in self.grid[x][y]:
```

Comment:
This needs to be `agent.pos is None`. The case when `agent.pos` is something that is not `None`, but is not `pos` either, means something is wrong. The agent might exist in both `self.grid[x][y]` and somewhere else.

Reply:
Good catch! I was thinking something similar, where agent.pos is simply set to something different, but dismissed it as unlikely. But agents being on grid and another space (e.g. network) is something I see happening. And in that case the agent.pos might as well be set to None by the network. 

I think the current implementation is the most defensive mechanism, not a bad thing.

Reply:
very good points but what you@are saying @Corvince make me think about something to resolve ambiguities, in a sense as you point out it doesn't make much sense to have a pos variable shared by all spaces, we should create a pos variable for each agent related to each space to which the agent belongs!

Reply:
By somewhere else, I meant `self.grid[x_other][y_other]`. `agent.pos` is not `None` happens when the modeler calls consecutive `place_agent`'s without ensuring that an agent is removed from the grid before a `place_agent` happens.

Reply:
yes, indeed this means that the current implementation is prone to errors